api/src/main/java/io/druid/data/input/impl/TimestampSpec.java

```diff
@@ -40,6 +40,10 @@
   // this value should never be set for production data
   private final DateTime missingValue;
 
+  // remember last value parsed
+  private Object lastTimeObject = null;
+  private DateTime lastDateTime = null;
```

Comment:
Another option would be to add an object:
class ParseContext
{
  Object timeVal;
  DateTime dateVal;
}

And instead of updating both values, we can just update the ParseContext member variable.

This way, it is thread-safe in terms of correctness, an if multiple threads modify the reference to the ParseContext, only performance gain would be limited.

Would that be a reasonable alternative?